### PR TITLE
Add Scp244Spawning Event

### PR DESCRIPTION
**Pull request by Rysik5318**

### CODE REVIEW

File: Exiled.Events/EventArgs/Map/Scp244SpawningEventArgs.cs [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        /// <param name="pickup">
+        ///     <inheritdoc cref="Pickup" />
+        /// </param>
+        /// <param name="scp244Pickup">
+        ///     <inheritdoc cref="Scp244Pickup" />
+        /// </param>
+        /// <param name="isAllowed">
+        ///     <inheritdoc cref="IsAllowed" />
+        /// </param>
+        public Scp244SpawningEventArgs(Room room, Pickup pickup, Scp244Pickup scp244Pickup, bool isAllowed = true)
```
Comment (VALERA771) [inline Exiled.Events/EventArgs/Map/Scp244SpawningEventArgs.cs:34]: What the reason to have both `Pickup` and `Scp244Pickup` (which inherits from `Pickup`)?

File: Exiled.Events/EventArgs/Map/Scp244SpawningEventArgs.cs [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+        /// <summary>
+        /// Gets the <see cref="API.Features.Pickups.Scp244Pickup"/> object that is spawning.
+        /// </summary>
+        public Scp244Pickup Scp244Pickup { get; }
+
+        /// <summary>
+        /// Gets or sets a value indicating whether or not the item can be spawning.
+        /// </summary>
+        public bool IsAllowed { get; set; } = true;
```
Comment (VALERA771) [inline Exiled.Events/EventArgs/Map/Scp244SpawningEventArgs.cs:60]:
```suggestion
        public bool IsAllowed { get; set; }
```

File: Exiled.Events/EventArgs/Map/Scp244SpawningEventArgs.cs (hunk 1 above)
Comment (Rysik5318) [reply Exiled.Events/EventArgs/Map/Scp244SpawningEventArgs.cs:34]: Isn't it better to give the developer the choice of what to use?
If he is interested in Pickup itself, let him use it.
If he is interested in Scp244Pickup, let him work with it.
For some reason there is a division by roles, for example: Scp106Role

Comment (VALERA771) [reply Exiled.Events/EventArgs/Map/Scp244SpawningEventArgs.cs:34]: You can get it like `Pickup = scp244Pickup`. And pickup and scp244Pickup is same anyways

Comment (Rysik5318) [reply Exiled.Events/EventArgs/Map/Scp244SpawningEventArgs.cs:34]: Hmmm, Okay

Comment (VALERA771) [reply Exiled.Events/EventArgs/Map/Scp244SpawningEventArgs.cs:34]: Or already inside contructor cast to Scp244Pickup

Comment (VALERA771) [reply Exiled.Events/EventArgs/Map/Scp244SpawningEventArgs.cs]: I think you don't really understandede me. Property should be but you remove constructor argument need to be removed

Comment (Rysik5318) [reply Exiled.Events/EventArgs/Map/Scp244SpawningEventArgs.cs]: I still don't understand you

Comment (VALERA771) [reply Exiled.Events/EventArgs/Map/Scp244SpawningEventArgs.cs]: You need to remove only construction argument

**Comment by Rysik5318** [comment]
@louis1706 добавьте так же тег "tested", я проверял, всё работает.

**Comment by louis1706** [comment]
> @louis1706 добавьте так же тег "tested", я проверял, всё работает.

Talk English

**Comment by Rysik5318** [comment]
@louis1706 also add a "tested" tag, I tested it, it works.

**Comment by louis1706** [comment]
you should add this back but you don't need to change the Ctor 

```csharp
        /// <summary>
        /// Gets the <see cref="API.Features.Pickups.Scp244Pickup"/> object that is spawning.
        /// </summary>
        public Scp244Pickup Scp244Pickup { get; }
```

**Comment by BoltonDev** [comment]
I made a [PR](https://github.com/Rysik5318/EXILED/pull/5) that target your branch @Rysik5318 (change Prefix to Transpiler)

**Comment by Misfiy** [comment]
@NaoUnderscore why was this marked as do not merge ?

**Pull request closed without merging**